### PR TITLE
[Angular] Upgrade to v17

### PR DESCRIPTION
**Pull request by addy-pathania**
- [x] This PR follows the [Contribution Guide](https://github.com/Sitecore/jss/blob/dev/CONTRIBUTING.md)
- [x] Changelog updated
- [ ] Upgrade guide entry added

## Testing Details

- [ ] Unit Test Added
- [x] Manual Test/Other (Please elaborate)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by art-alexeyenko — CHANGES_REQUESTED** [review]
Great job as always, Addy.
I have just one concern here: how would we go about the upgrade guide? It won't be extensive, but we need to leave some pointers for people updating v21 apps.

File: packages/sitecore-jss-angular/package.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
     "generate-docs": "npx typedoc --plugin typedoc-plugin-markdown --readme none --out ../../ref-docs/sitecore-jss-angular src/public_api.ts --githubPages false"
   },
   "engines": {
-    "node": ">=18"
+    "node": ">=20"
```
Comment (illiakovalenko) [inline packages/sitecore-jss-angular/package.json:14]: We can't change it to Node 20, since other packages continue to use Node 18

Comment (addy-pathania) [reply packages/sitecore-jss-angular/package.json:14]: make sense.

File: packages/sitecore-jss-angular/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@sitecore-jss/sitecore-jss-angular",
-  "version": "21.7.1",
+  "version": "22.1.0-canary.43",
```
[the remark this answers is not in the document]
Comment (addy-pathania) [reply packages/sitecore-jss-angular/package.json:3]: Yes you're correct, it should but i forgot to run the hotfix-prep script before starting to work on this.
I will not merge this until i update the main release branch and then later fix the merge conflicts.

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,26 @@ Our versioning strategy is as follows:
...
 
+## 21.8.0
+
+### 🛠 Breaking Changes
+
+`[sitecore-jss-angular]` `[sitecore-jss-schematics]` `[templates/angular]` Upgrade Angular to v17 ([#1939](https://github.com/Sitecore/jss/pull/1939))
+  - Updated Angular and core dependencies to ~17.3.11
+  - Updated Typescript to ~5.2.2
+  - Updated import statements from zone.js/dist/zone-node to zone.js
+      Use _'zone.js'_ instead of _'zone.js/dist/zone-node'_
```
Comment (art-alexeyenko) [inline CHANGELOG.md:20]: IMO it would be cleaner if you put these instructions into docs/upgrades/../21.8/md and have a link to it here in release notes. Something like "You can find the uprade instruction in [here]".
Please also add the link to angular's v16->v17 upgrade guide for extra reference in our upgrade guide.

**Review by art-alexeyenko — CHANGES_REQUESTED** [review]

File: packages/sitecore-jss-angular/package.json [hunk 4]
```diff
@@ -23,38 +23,38 @@
...
+    "@angular/cli": "~17.3.10",
+    "@angular/common": "~17.3.12",
+    "@angular/compiler": "~17.3.12",
+    "@angular/compiler-cli": "~17.3.12",
+    "@angular/core": "~17.3.12",
+    "@angular/platform-browser": "~17.3.12",
+    "@angular/platform-browser-dynamic": "~17.3.12",
+    "@angular/router": "~17.3.12",
+    "@types/jasmine": "^5.1.4",
+    "@types/node": "^20.14.10",
```
Comment (illiakovalenko) [inline packages/sitecore-jss-angular/package.json:37]: Since we continue to use node 18, we need to revert this change

File: CHANGELOG.md (hunk 3 above)
Comment (illiakovalenko) [reply CHANGELOG.md:20]: @art-alexeyenko We can't really put these instructions under 21.8.md file, since we expect to publish migration guide to official doc side

File: packages/sitecore-jss-angular/package.json (hunk 4 above)
Comment (illiakovalenko) [reply packages/sitecore-jss-angular/package.json:37]: We can leave it, since it's a dev dependency

File: CHANGELOG.md (hunk 3 above)
Comment (art-alexeyenko) [reply CHANGELOG.md]: @illiakovalenko does it make sense to publish a two-point guide to doc site however? Our general approach is to have upgrade instructions available through github - in this case they will be.

Comment (illiakovalenko) [reply CHANGELOG.md]: @art-alexeyenko Let's discuss more later, I will raise a question regarding all "migration" related files

Comment (art-alexeyenko) [reply CHANGELOG.md]: DIscussed this more.
Let's remove the instructions from github and post them on doc site as usual.

**Review by art-alexeyenko — APPROVED** [review]

**Review by illiakovalenko — APPROVED** [review]
👍
